### PR TITLE
Lots3

### PR DESCRIPTION
**Pull request by AlexRedisson18**
## What was done?
- add S3

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,6 +9,9 @@ gem 'pg'
 gem 'pry-rails'
 gem 'rails', '~> 6.0.2', '>= 6.0.2.2'
 gem 'slim-rails'
+gem 'carrierwave', '~> 2.0'
+gem 'mini_magick'
+gem "fog-aws"
```
Comment (akurashev) [inline Gemfile]:
```suggestion
gem 'fog-aws'
```

File: app/controllers/lots_controller.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 class LotsController < ApplicationController
-  respond_to :json, only: %i[create]
```
Comment (akurashev) [inline app/controllers/lots_controller.rb:2]: Hmm... Why is that removed?

File: app/controllers/lots_controller.rb [hunk 3]
```diff
@@ -41,7 +40,10 @@ def lot_params
       :state,
       :price,
       :category_id,
-      interesting_category_ids: []
+      { interesting_category_ids: [] },
+      :image,
+      { images: [] },
+      :image_cache
```
Comment (akurashev) [inline app/controllers/lots_controller.rb]: Keep hash params at the end

```suggestion
      :image,
      :image_cache,
      interesting_category_ids: [],
      images: []
```

File: app/models/lot.rb [hunk 4]
```diff
@@ -27,4 +27,7 @@ def no_price?
   def no_categories?
     interesting_categories.empty?
   end
+
+  mount_uploader :image, ImageUploader
+  mount_uploaders :images, ImageUploader
```
Comment (akurashev) [inline app/models/lot.rb]: Move this up. Place it somewhere around relations/validations/scopes

**Review by akurashev — CHANGES_REQUESTED** [review]

File: app/controllers/lots_controller.rb (hunk 2 above)
Comment (AlexRedisson18) [reply app/controllers/lots_controller.rb:2]: i tried to remove it - and nothing changed
i think it doesnt work

**Review by akurashev — CHANGES_REQUESTED** [review]
One comment left. And fix conflicts

**Review by akurashev — APPROVED** [review]

**Pull request merged**